source/extensions/quic_listeners/quiche/active_quic_listener.h

```diff
@@ -18,55 +18,60 @@ namespace Quic {
 
 // QUIC specific UdpListenerCallbacks implementation which delegates incoming
 // packets, write signals and listener errors to QuicDispatcher.
-class ActiveQuicListener : public Network::UdpListenerCallbacks,
-                           public Server::ConnectionHandlerImpl::ActiveListenerImplBase,
+class ActiveQuicListener : public Envoy::Server::ActiveUdpListenerBase,
                            Logger::Loggable<Logger::Id::quic> {
 public:
   // TODO(bencebeky): Tune this value.
   static const size_t kNumSessionsToCreatePerLoop = 16;
 
-  ActiveQuicListener(Event::Dispatcher& dispatcher, Network::ConnectionHandler& parent,
-                     Network::ListenerConfig& listener_config, const quic::QuicConfig& quic_config,
-                     Network::Socket::OptionsSharedPtr options,
+  ActiveQuicListener(uint32_t worker_id, Event::Dispatcher& dispatcher,
+                     Network::ConnectionHandler& parent, Network::ListenerConfig& listener_config,
+                     const quic::QuicConfig& quic_config, Network::Socket::OptionsSharedPtr options,
+                     bool kernel_worker_routing,
                      const envoy::config::core::v3::RuntimeFeatureFlag& enabled);
 
-  ActiveQuicListener(Event::Dispatcher& dispatcher, Network::ConnectionHandler& parent,
-                     Network::SocketSharedPtr listen_socket,
+  ActiveQuicListener(uint32_t worker_id, Event::Dispatcher& dispatcher,
+                     Network::ConnectionHandler& parent, Network::SocketSharedPtr listen_socket,
                      Network::ListenerConfig& listener_config, const quic::QuicConfig& quic_config,
-                     Network::Socket::OptionsSharedPtr options,
+                     Network::Socket::OptionsSharedPtr options, bool kernel_worker_routing,
                      const envoy::config::core::v3::RuntimeFeatureFlag& enabled);
 
   ~ActiveQuicListener() override;
 
   void onListenerShutdown();
+  uint64_t eventLoopsWithBufferedChlos() const { return event_loops_with_buffered_chlo_; }
```

Comment:
If this is only for testing can you postfix with `ForTest` and `for_test_`? I don't see any prod use of this.